ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py

```diff
@@ -369,9 +366,27 @@ def wan_ha_flow_tracker_name(self: SharedUtils) -> str:
         """
         return "WAN-FLOW-TRACKER"
 
+    # TODO should we keep this or change it?
     @cached_property
     def is_cv_pathfinder_edge_or_transit(self: SharedUtils) -> bool:
         """
         Return True is the current wan_mode is cv-pathfinder and the device is either an edge or a transit device
         """
-        return self.wan_mode == "cv-pathfinder" and self.cv_pathfinder_role in ["edge", "transit region"]
+        return self.wan_mode == "cv-pathfinder" and self.is_wan_client
+
+    @cached_property
+    def cv_pathfinder_role(self: SharedUtils) -> str | None:
+        """
+        Return the CV Pathfinder role based on the wan_role (server or client) and the
+        node.cv_pathfinder_transit_mode key.
+        """
+        if self.wan_mode != "cv-pathfinder" or not self.is_wan_router:
+            return None
+
+        if self.is_wan_server:
+            return "pathfinder"
+        elif self.cv_pathfinder_transit_mode == "none":
```

Comment:
Something about this match makes me feel uncomfortable. Suggested None value for the `"none"` case.